src/context_impl.hpp

```diff
@@ -70,12 +70,12 @@ class ContextImpl : public virtual Context {
 	libm2k::analog::DMM* getDMM(std::string);
 	std::vector<libm2k::analog::DMM*> getAllDmm();
 
-
 	std::vector<std::string> getAvailableContextAttributes();
 	std::string getContextAttributeValue(std::string attr);
 	std::string getContextDescription();
 	std::string getSerialNumber();
 	std::unordered_set<std::string> getAllDevices();
+	void logAllAttributes() override;
```

Comment:
can this method be tagged `const`?

Reply:
I think that const for void function has no utility.

Reply:
it's not about the return type, it's about the const correctness of the api. Even if a method is void it can still alter the internal state of the object/ or not. Example `void clear()` of a `std::vector`. In our case here: does the logging of all the attributes of the context, alter its internal state in any way?

Reply:
Another example:
```cpp
class Whatever {
public:
  void inspect() const;   // This member promises NOT to change *this
  void mutate();          // This member function might change *this
};
```

Reply:
Modified.